offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/offer/create/SelectItemsView.groovy

```diff
@@ -69,6 +74,27 @@ class SelectItemsView extends VerticalLayout{
     TextField amountMetabolomicAnalysis
     TextField amountDataStorage
 
+    /**
+     * Contains regex for filtering the different product types
+     *
+     * This enum stores the regex for validating a user input and differences between an atomic product
+     * which only allows integer input and a partial product which allows double input.
+     *
+     */
+    enum ProductTypeRegex {
+        ATOMIC("^[0-9]+\$"),
+        PARTIAL("[-]?[0-9]*\\.?[0-9]+"),
+
+        private String productType
+
+        ProductTypeRegex(String productType) {
+            this.productType = productType;
+        }
+
+        String getType() {
+            return this.type;
```

Comment:
This confused me, because of the naming. You are not creating a type, but passing the regex value as parameter. The type is implicit by the enum (that is why u have defined it in the first place ;)

```suggestion
        private String regex

        ProductTypeRegex(String regex) {
            this.regex = regex;
        }

        String getRegex() {
            return this.regex;
```